### PR TITLE
CA-290891: do not call fdatasync on pipes/sockets

### PR DESCRIPTION
**Pull request by edwintorok**
fdatasync will return EINVAL on special files such as pipes, and sockets.
We only want fdatasync if we are writing to a file or block device.

### CODE REVIEW

File: src/channels.ml [hunk 1]
```diff
@@ -77,12 +77,20 @@ let rec sendfile from_fd to_fd len =
     (fun from_fd ->
       with_blocking_fd to_fd
         (fun to_fd ->
+          Lwt_unix.LargeFile.fstat to_fd
+          >>= fun stat ->
+          let maybe_fdatasync () =
+            match stat.Lwt_unix.LargeFile.st_kind with
+            | Unix.S_REG | Unix.S_BLK ->
```
Comment (gaborigloi) [inline src/channels.ml]: So I assume only these two support synchronization

Comment (lindig) [reply src/channels.ml]: Why is `maybe_fdatasync` defined within the sequence of binds? I would have defined it a bit more independently. However, this should have no impact on functionality.

Comment (edwintorok) [reply src/channels.ml]: It needed access to `stat` and `to_fd`. I moved it to the toplevel now, agreed that its cleaner that way.

**Comment by lindig** [comment]
I will port this to the `0.11-lcm` branch.

**Pull request merged**